src/libcode/vx_data2d_grib2/data2d_grib2.cc

```diff
@@ -1294,45 +1295,101 @@ void MetGrib2DataFile::read_grib2_grid( gribfield *gfld) {
+            break;
+
+         case 6:
+            laea.radius_km            = 6371.2290;
+            laea.equatorial_radius_km = 0;
+            laea.polar_radius_km      = 0;
+            laea.is_sphere            = true;
+            break;
+
+         case 7:
+            laea.radius_km            = 0;
+            laea.equatorial_radius_km = scaled2dbl(gfld->igdtmpl[3], gfld->igdtmpl[4]) / 1000.0;
+            laea.polar_radius_km      = scaled2dbl(gfld->igdtmpl[5], gfld->igdtmpl[6]) / 1000.0;
+            laea.is_sphere            = false;
+            break;
+
+         default:
+            mlog << Error << "\nMetGrib2DataFile::read_grib2_grid() -> "
+                 << "unsupported earth shape value of " << earth_shape_int << "!\n\n";
+            exit(1);
       }
-      laea.nx           = p[7];
-      laea.ny           = p[8];
-      laea.lat_first    = (double)p[9] / 1000000.0; 
-      // TODO: Suspect a bug in g2clib
-      // laea.lon_first = -1.0*rescale_lon( (double)p[10] / 1000000.0 );
+
+      //
+      // MET#2565:
+      //   Fix a bug when parsing laea.lon_first and laea.central_lon
+      //   in gfld->igdtmpl[10] and gfld->igdtmpl[12] for GRIB2 UKV data.
+      //   These longitudes are negative which the GRIB2C library does not
+      //   parse properly. The parse_int4() function is a workaround to
+      //   reprocesses the data as unsigned chars and check for negative
+      //   values, mimicing logic from the int4() function in wgrib2.
       //
-      // grib_dump: longitudeOfFirstGridPointInDegrees = -17.1171;
-      // wgrib2: Lon1 2164.600777
-      //    3-43=129,5,47,201
-      // g2clib-1.6.0 grid_templates.h
-      //    3.140: Lambert Azimuthal Equal Area Projection
-      // BAD?   {140, 17, 0, {1,1,4,1,4,1,4,4,4,-4,4,4,4,1,4,4,1} },
-      // FIX?   {140, 17, 0, {1,1,4,1,4,1,4,4,4,-4,4,-4,4,1,4,4,1} },
-      cout << "TODO LAEA p[10] should be 17.1171 = "<< (double)p[10] / 1000000.0 << "\n";
-      laea.lon_first    = 17.1171;
-      laea.standard_lat = (double)p[11] / 1000000.0; 
-      laea.central_lon  = -1.0*rescale_lon( (double)p[12] / 1000000.0 );
-      laea.dx_km        = (double)p[14] / 1000000.0;
-      laea.dy_km        = (double)p[15] / 1000000.0;
+      //   The grib_dump utility already handles negative longitudes well.
+      //   The wgrib2 utility does NOT when populating gfld->igdtmpl.
+      //   However, the wgrib2 -get_byte and -get_int options do work,
+      //   as shown below:
+      //      wgrib2 ukv_agl_temperature_1.5_12.grib2 \
+      //        -get_byte 3 43 4 -get_int 3 43 1 \
+      //        -get_byte 3 51 4 -get_int 3 51 1
+      //
+      //   The longitudes are correctly parsed as -17.117129 and -2.5, respectively.
+      //   1:0:3-43=129,5,47,201:3-43=-17117129:3-51=128,38,37,160:3-51=-2500000
+      //
+
+      laea.nx           = gfld->igdtmpl[7];
+      laea.ny           = gfld->igdtmpl[8];
+      laea.lat_first    = (double)gfld->igdtmpl[9] / 1000000.0;
```

Comment:
I'm not sure if it's worth it, particularly with the time crunch, but I was wondering if we should consider creating a variable for the 1000000.0 value, which is reused?

Reply:
Good point Julie. Thanks for mentioning that. I'd like to not make a change at this time because it would apply to about 40 lines of code in this file. I'd rather limit the changes to what's explicitly necessary for this issue during the official release testing time period to limit the lines of code I touch. But this would be good to consider as part of the changes to the vx_grid library's use of the proj library in the future.